runtime/doc/lua.txt

```diff
@@ -827,6 +827,61 @@ vim.schedule({callback})				*vim.schedule()*
         Schedules {callback} to be invoked soon by the main event-loop. Useful
         to avoid |textlock| or other temporary restrictions.
 
+
+vim.defer_fn({fn}, {timeout})                                    *vim.defer_fn*
+    Defers calling {fn} until {timeout} ms passes.  Use to do a one-shot timer
+    that calls {fn}.
+
+    Parameters: ~
+        {fn}        Callback to call once {timeout} expires
+        {timeout}   Time in ms to wait before calling {fn}
+
+    Returns: ~
+        |vim.loop|.new_timer() object
+
+vim.wait({time}, {callback} [, {interval}])                         *vim.wait()*
+        Wait for {time} in milliseconds until {callback} returns `true`.
+
+        Executes {callback} at approximately {interval} milliseconds
+        (default 200). Nvim still processes other events during this time.
```

Comment:
isn't it weird to have a default for interval ? I would be surprised by the behavior I guess.

Reply:
what if we set interval to <= 0 ?

Reply:
> isn't it weird to have a default for interval ? I would be surprised by the behavior I guess.

I think so too. Interval argument should be required.

Reply:
Why is default 200ms?

Reply:
Why is default 200ms?

Reply:
> what if we set interval to 0 ?

Error occurs.
https://github.com/neovim/neovim/pull/12348/files#diff-619381246c51a32ff4cf5de3e7cd627cR301

Reply:
the check is only for <0 not if it equals 0. This should show the same behavior as the libuv one but we shouldn't ask user to go thriug the libuv doc and explicitly state the behavior instead.

Reply:
Nevermind the question about the prototype, it just follows the f_wait convention.

Reply:
Oh sorry, I miss it.